File: web/src/services/api/webhook.ts
import apiClient from "./client";

export type WebhookConfig = {
    id?: number;
    tenant_id?: number;
    platform: string;
    webhook_url: string;
    enabled: boolean;
    cooldown_minutes?: number;
};

export type WebhookLogItem = {
    id: number;
    tenant_id?: number;
    platform: string;
    channel_id?: number;
    channel_name?: string;
    model_name: string;
    status: string;
    message: string;
    success: boolean;
    response_body: string;
    cooldown_skipped: boolean;
    created_at: string;
};

export type TestSendInput = {
    platform: string;
    message: string;
};

export type TestSendResult = {
    success: boolean;
    error?: string;
};

export type SaveWebhookConfigInput = Pick<WebhookConfig, "platform"> & Partial<Omit<WebhookConfig, "platform">>;

export async function listWebhookConfigs() {
    const res = await apiClient.get("/admin/notifications/webhooks");
    return res.data.data as WebhookConfig[];
}

export async function saveWebhookConfig(input: SaveWebhookConfigInput) {
    const res = await apiClient.put("/admin/notifications/webhooks", input);
    return res.data.data as WebhookConfig;
}

export async function testWebhookSend(input: TestSendInput) {
    const res = await apiClient.post("/admin/notifications/webhooks/test", input);
    return res.data.data as TestSendResult;
}

export async function listWebhookLogs(limit?: number) {
    const res = await apiClient.get("/admin/notifications/webhook-logs", { params: { limit } });
    return res.data.data as WebhookLogItem[];
}

File: backend/model/webhook.go
package model

type WebhookConfig struct {
	BaseModel
	TenantID        uint   `gorm:"index;not null" json:"tenant_id"`
	Platform        string `gorm:"size:50;not null" json:"platform"`
	WebhookURL      string `gorm:"size:500;not null" json:"webhook_url"`
	Enabled         bool   `gorm:"default:true" json:"enabled"`
	CooldownMinutes int    `gorm:"default:10" json:"cooldown_minutes"`
}

func (WebhookConfig) TableName() string { return "webhook_configs" }

type WebhookLog struct {
	BaseModel
	TenantID        uint   `gorm:"index;not null" json:"tenant_id"`
	Platform        string `gorm:"size:50;not null" json:"platform"`
	ChannelID       uint   `gorm:"index" json:"channel_id"`
	ChannelName     string `gorm:"size:100" json:"channel_name"`
	ModelName       string `gorm:"size:100" json:"model_name"`
	Status          string `gorm:"size:50;not null" json:"status"`
	Message         string `gorm:"type:text" json:"message"`
	Success         bool   `gorm:"not null;default:false" json:"success"`
	ResponseBody    string `gorm:"type:longtext" json:"response_body"`
	CooldownSkipped bool   `gorm:"default:false" json:"cooldown_skipped"`
}

func (WebhookLog) TableName() string { return "webhook_logs" }
